core/src/test/scala/unit/kafka/utils/timer/MockTimer.scala

```diff
@@ -17,12 +17,15 @@
 package kafka.utils.timer
 
 import org.apache.kafka.server.util.MockTime
+import org.apache.kafka.server.util.timer.{Timer, TimerTask, TimerTaskEntry}
 
 import scala.collection.mutable
 
 class MockTimer(val time: MockTime = new MockTime) extends Timer {
 
-  private val taskQueue = mutable.PriorityQueue[TimerTaskEntry]()(Ordering[TimerTaskEntry].reverse)
+  private val taskQueue = mutable.PriorityQueue.empty[TimerTaskEntry](new Ordering[TimerTaskEntry] {
+    override def compare(x: TimerTaskEntry, y: TimerTaskEntry): Int = java.lang.Long.compare(x.expirationMs, y.expirationMs)
```

Comment:
This can be removed after we add compare method to TimerTaskEntry

Reply:
I cannot use `Ordering` in Java. This is why I change it this way.

Reply:
The Java equivalent of scala `Ordering` could be `Comparable`. That will force you to implement a compareTo function for TimerTaskEntry.

Reply:
Right but I don’t think that the scala priority queue accepts this. This is why I do it inline here.